doc/source/whatsnew/v0.21.0.txt

```diff
@@ -328,6 +328,13 @@ Previously, :func:`to_datetime` did not localize datetime ``Series`` data when `
 
 Additionally, DataFrames with datetime columns that were parsed by :func:`read_sql_table` and :func:`read_sql_query` will also be localized to UTC only if the original SQL columns were timezone aware datetime columns.
 
+.. _whatsnew_0210.api.restricting_dateoffset_keywords
+
+Restricting DateOffset Keywords
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, ``DateOffset`` subclasses allowed arbitrary keyword arguments which could lead to unexpected behavior.  Now, subclasses :class:`Week`, :class:`WeekOfMonth`, :class:`LastWeekOfMonth`, :class:`QuarterOffset`, :class:`QuarterEnd`, :class:`QuarterBegin`, :class:`BQuarterEnd`, and :class:`BQuarterBegin`, will allow only specific arguments.  (:issue:`17176`).
```

Comment:
aren't Month/MonthEnd included here? why not *all* classes?

Reply:
> aren't Month/MonthEnd included here?

Not as of now, no.

> why not _all_ classes?

Wanted to change them a couple at a time, err on the side of doing too little.  Now that #17486 is resolved I can push forward on a few more.  You want to just get them all in one go?

Reply:
yes for this type of change am ok with doing it in one fell swoop. On the whatsnew just highlite the change generally (e.g. give a single example), then list the classes that are affected (or if its all, then just say all). imagine a user who just 'uses' pandas. They want to quickly get to: does this change affect me, if so, what do I need to do to fix.